File: front-end/src/app/documento-dispensado/models/documento-dispensado-para-pesquisar-resource.ts
import { CustomPageResource } from '@shared/models/custom-page-resource';

import { ClienteParaSelecaoResource } from '@app/cliente/models/cliente-para-selecao-resource';
import { DocumentoParaSelecaoResource } from '@app/documento/models/documento-para-selecao-resource';
import { GrupoDocumentoParaSelecaoResource } from '@app/grupo-documento/models/grupo-documento-para-selecao-resource';
import { SituacaoDocumentoDispensado } from '@app/documento-dispensado/models/situacao-documento-dispensado';
import { DocumentoDispensadoResultadoPesquisaResource } from '@app/documento-dispensado/models/documento-dispensado-resultado-pesquisa-resource';

export class DocumentoDispensadoParaPesquisarResource {

  public acoes: Acoes;
  public dadosFormulario: DadosFormulario;
  public dadosParaFormulario: DadosParaFormulario;
  public resultadoPesquisa: CustomPageResource<DocumentoDispensadoResultadoPesquisaResource>;
}

export class Acoes {

  public podeCriar: boolean;
  public podeGerarRelatorioExcel: boolean;

  constructor(values: Object) {
    Object.assign(this, values);
  }
}

export class DadosFormulario {

  public clientes: Array<ClienteParaSelecaoResource>;
  public dataBaseFim: Date;
  public dataBaseInicio: Date;
  public documentos: Array<DocumentoParaSelecaoResource>;
  public grupoDocumentos: Array<GrupoDocumentoParaSelecaoResource>;
  public situacaoDocumentoDispensados: Array<SituacaoDocumentoDispensado>;
}

export class DadosParaFormulario {

  public clientes: Array<ClienteParaSelecaoResource>;
  public documentos: Array<DocumentoParaSelecaoResource>;
  public grupoDocumentos: Array<GrupoDocumentoParaSelecaoResource>;
  public situacaoDocumentoDispensados: Array<SituacaoDocumentoDispensado>;
}

File: backend/rest/src/main/java/com/github/thiagogarbazza/training/springangular/rest/controller/documentodispensado/resource/DocumentoDispensadoParaPesquisarResource.java
package com.github.thiagogarbazza.training.springangular.rest.controller.documentodispensado.resource;

import com.github.thiagogarbazza.training.springangular.core.documentodispensado.SituacaoDocumentoDispensado;
import com.github.thiagogarbazza.training.springangular.rest.comum.consulta.CustomPageResource;
import com.github.thiagogarbazza.training.springangular.rest.controller.cliente.resource.ClienteParaSelecaoResource;
import com.github.thiagogarbazza.training.springangular.rest.controller.documento.resource.DocumentoParaSelecaoResource;
import com.github.thiagogarbazza.training.springangular.rest.controller.grupodocumento.resource.GrupoDocumentoParaSelecaoResource;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.time.LocalDate;
import java.util.Collection;

import static lombok.AccessLevel.PRIVATE;

@Getter
@Setter
@Builder
@ToString
@NoArgsConstructor(access = PRIVATE)
@AllArgsConstructor(access = PRIVATE)
public class DocumentoDispensadoParaPesquisarResource {

  private Acoes acoes;
  private DadosFormulario dadosFormulario;
  private DadosParaFormulario dadosParaFormulario;
  private CustomPageResource<DocumentoDispensadoResultadoPesquisaResource> resultadoPesquisa;

  @Getter
  @Setter
  @Builder
  @ToString
  @NoArgsConstructor(access = PRIVATE)
  @AllArgsConstructor(access = PRIVATE)
  public static class Acoes {

    private boolean podeCriar;
    private boolean podeGerarRelatorioExcel;
  }

  @Getter
  @Setter
  @Builder
  @ToString
  @NoArgsConstructor(access = PRIVATE)
  @AllArgsConstructor(access = PRIVATE)
  public static class DadosFormulario {

    private Collection<ClienteParaSelecaoResource> clientes;
    private LocalDate vigenciaFim;
    private LocalDate vigenciaInicio;
    private Collection<DocumentoParaSelecaoResource> documentos;
    private Collection<GrupoDocumentoParaSelecaoResource> grupoDocumentos;
    private Collection<SituacaoDocumentoDispensado> situacaoDocumentoDispensados;
  }

  @Getter
  @Setter
  @Builder
  @ToString
  @NoArgsConstructor(access = PRIVATE)
  @AllArgsConstructor(access = PRIVATE)
  public static class DadosParaFormulario {

    private Collection<ClienteParaSelecaoResource> clientes;
    private Collection<DocumentoParaSelecaoResource> documentos;
    private Collection<GrupoDocumentoParaSelecaoResource> grupoDocumentos;
    private Collection<SituacaoDocumentoDispensado> situacaoDocumentoDispensados;
  }
}
